build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
```

Comment:
As long as we can return a `Gem::Version` here, I think our confine can be rewrtiten as:

```ruby
confine :operatingsystem => :windows, :powershell_version => Gem::Version.new('5.0.10514.6')
```

But I've only returned basic types like strings and numerics from facts previously, so double check that Facter will preserve the `Gem::Version` type, and doesn't stringify it.

If it is stringifying it, we can rewrite the confine another way to turn the fact into a `Gem::Version` for the sake of comparison.